clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -1,13 +1,92 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+import com.google.gson.JsonArray;
 import org.junit.Test;
 
 import static org.junit.Assert.*;
 
 public class HubConnectionTest {
-    // Coming soon!
     @Test
-    public void testEmptyCollection() {
+    public void checkHubConnectionState() throws InterruptedException {
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("test", mockTransport);
+            hubConnection.start();
+            assertTrue(hubConnection.connected);
+
+            hubConnection.stop();
+            assertFalse(hubConnection.connected);
+    }
+
+    @Test
+    public void SendWithNoParamsTriggersOnHandler() throws InterruptedException {
+        TestObject obj = new TestObject();
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("placeholder", mockTransport);
+
+        Action callback = (param)-> {
+            assertEquals(0, obj.counter);
+            obj.incrementCounter();
+        };
+        hubConnection.On("inc", callback);
+
+        hubConnection.start();
+        hubConnection.send("inc");
+
+        // Confirming that our handlers was called and that the counter property was incremented.
+        assertEquals(1, obj.counter);
+    }
+
+    @Test
+    public void SendWithParamTriggersOnHandler() throws InterruptedException {
+        TestObject obj = new TestObject();
```

Comment:
This can just be a string

Reply:
So it actually can't just be a string here. It would have to be declared as final, which won't work for this, or as an `AtomicReference<String>`.  I just oped to use the `TestObject ` instead